ui/packages/catalog-ui-search/src/main/webapp/react-component/filter/filter-attribute.js

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react'
+import withListenTo from '../../react-component/container/backbone-container'
+import MarionetteRegionContainer from '../../react-component/container/marionette-region-container'
+import styled from 'styled-components'
+const DropdownView = require('../../component/dropdown/dropdown.view.js')
+const properties = require('../../js/properties.js')
+const metacardDefinitions = require('../../component/singletons/metacard-definitions.js')
+const Root = styled.div`
+  display: inline-block;
+  vertical-align: middle;
+  margin-right: ${({ theme }) => theme.minimumSpacing};
+  height: ${({ theme }) => theme.minimumButtonSize};
+  line-height: ${({ theme }) => theme.minimumButtonSize};
+  intrigue-dropdown.is-editing .dropdown-text {
+    width: auto;
+    max-width: 300px;
+  }
+`
+const FilterAttribute = withListenTo(
+  class FilterAttribute extends React.Component {
+    component
+    constructor(props) {
+      super(props)
+      this.state = {
```

Comment:
I think you can just pull things from props whenever you need `editing` and `attribute`. Same thing for filter-input. You could make this be a functional component rather than a class component if you don't end up using state